### ISSUE TITLE
loss:nan grdn:nan How to solve the gradient explosion problem in PI05 training?

### ISSUE THREAD

**Issue by Lilgeneric** [huggingface/lerobot#2465]
When training Pi05 using Lerobot, has anyone encountered a situation where gradients explode immediately after training? Errors occur when the batch_size is set to 64 or 32. How can this be resolved? 

Below are my training commands and error logs.

python src/lerobot/scripts/lerobot_train.py --dataset.repo_id=aa_merged280 --policy.type=pi05 \
--output_dir=./outputs/pi05_training2 --job_name=pi05_training2 \
--policy.pretrained_path=lerobot/pi05_base --policy.compile_model=true \
--policy.gradient_checkpointing=true --wandb.enable=true --policy.dtype=bfloat16 \
--steps=100000 --policy.device=cuda --batch_size=32 --policy.push_to_hub=false

INFO 2025-11-17 22:07:40 ot_train.py:351 step:200 smpl:6K ep:9 epch:0.03 loss:nan grdn:nan lr:2.5e-06 updt_s:4.478 data_s:0.038
WARNING 2025-11-17 22:07:40 db_utils.py:141 WandB logging of key "loss_per_dim" was ignored as its type "<class 'list'>" is not handled by this wrapper.
INFO 2025-11-17 22:22:38 ot_train.py:351 step:400 smpl:13K ep:18 epch:0.06 loss:nan grdn:nan lr:7.5e-06 updt_s:4.458 data_s:0.022
WARNING 2025-11-17 22:22:38 db_utils.py:141 WandB logging of key "loss_per_dim" was ignored as its type "<class 'list'>" is not handled by this wrapper.
INFO 2025-11-17 22:37:34 ot_train.py:351 step:600 smpl:19K ep:27 epch:0.10 loss:nan grdn:nan lr:1.3e-05 updt_s:4.456 data_s:0.022
WARNING 2025-11-17 22:37:34 db_utils.py:141 WandB logging of key "loss_per_dim" was ignored as its type "<class 'list'>" is not handled by this wrapper.
INFO 2025-11-17 22:52:31 ot_train.py:351 step:800 smpl:26K ep:36 epch:0.13 loss:nan grdn:nan lr:1.8e-05 updt_s:4.456 data_s:0.022
WARNING 2025-11-17 22:52:31 db_utils.py:141 WandB logging of key "loss_per_dim" was ignored as its type "<class 'list'>" is not handled by this wrapper.
INFO 2025-11-17 23:07:29 ot_train.py:351 step:1K smpl:32K ep:45 epch:0.16 loss:nan grdn:nan lr:2.3e-05 updt_s:4.459 data_s:0.022

**Comment by liuslevis** [comment]
Hi, I'm facing same issue, but I succeed after retry.

It turns out the `clip_grad_norm` did not work properly and it sometimes return `grad_nrom` in a value much greater than `grad_clip_norm` i set (e.g. 1.0). You could print the grad for debug:

```
# Clip gradients if specified
    if grad_clip_norm > 0:
        grad_norm = accelerator.clip_grad_norm_(policy.parameters(), grad_clip_norm)
        logging.info(f"XJ debug grad_norm {grad_norm} grad_clip_norm {grad_clip_norm} params: {policy.parameters()}")
```

I assume there are some bug in `accelerator.clip_grad_norm_()`.


Maybe you could try print the grad / batch_size=1 to debug.
